### ISSUE TITLE
tcp.accept can sometimes accept GHOST SOCKETs (previous closed remote connection).

### ISSUE THREAD

**Issue by jakitliang** [lunarmodules/luasocket#410]
You can test it with your local `irb` or `python` or `nodejs` as a client.

```ruby
require 'socket'
c = TCPSocket.new 127.0.0.1, 3000
c.write '123'
c.close()
```

```lua
--- Socket = require 'luasocket'
--- local server =  Socket.tcp()
server:bind('127.0.0.1', 3000)
server:listen()

while true do
  r, w, err IO.select({server}, nil, 1)

  if #r > 0
    if r[1] == server then
      local client = server:accpet() -- previous closed client is successfully accepted but this client had closed by ruby already.
      -- You accept A ZOMBIE SOCKET!!!
      msg1, err, msg2 = client.receive()
      client.send(msg1 or msg2)
      client.close()
    end
  end
end
```

**Client only connect The SERVER by 1 TIME.**

**But Server Would Accept more Than 1 TIME**

I mean:

```
Server -> select()
Client -> connect()
Server -> accept
Client -> send() -> close()
Server -> receive() -> send(response) -> close()
Server -> select()
Server -> accept() -- HERE is the problem, now no client is doing connect().
-- But a GHOST socket being accept()
```

`Express.js` and `Ruby on Rails` Don't have such problem.

Only `luasocket` have this bug.

**Comment by alerque** [comment]
I think it is possible the very old but only recently merged PR #81 might address this issue. Can you check it out again with the current SCM version?

**Comment by jakitliang** [comment]
I review this commit and found that there did add a retry by `continue`.

But really sorry that I can't recheck this issue because my project had replace `luasocket` into another module [Socket.lua](https://github.com/jakitliang/Socket.lua).

Sorry again and hope `luasocket` well in the future!

Best wishes!
